src/box/lua/load_cfg.lua

```diff
@@ -187,6 +188,56 @@ local default_cfg = {
     txn_isolation         = "best-effort",
 }
 
+-- We need to track cfg changes done through API of distinct modules (log.cfg of
+-- log module for example). We cannot use just box.cfg because it is not
+-- available before box.cfg() call and other modules can be configured before
+-- this moment.
+local pre_load_cfg = table.copy(default_cfg)
+
+-- On first box.cfg{} we need to know options that were already configured
+-- in standalone modules (like log module). We should not apply env vars
+-- for these options. pre_load_cfg is not suitable for this purpose because
+-- of nil values.
+local pre_load_cfg_is_set = {}
+
+local replication_sync_timeout_brief = [[
+Sets the default value for box.cfg.replication_sync_timeout.
+Old is 300 seconds, new is 0 seconds. New behaviour makes
+box.cfg{replication = ...} call exit without waiting for
+synchronisation with all the remote nodes. This means that the node
+might be in 'orphan' state for some time after the box.cfg{} call
+returns.
+
+Set before first box.cfg{} call in order for the option to take effect.
+]]
+
+-- A list of box.cfg options whose defaults are managed by compat.
+local compat_options = {
+    {
+        name = 'replication_sync_timeout',
+        brief = replication_sync_timeout_brief,
+        oldval = 300,
+        newval = 0,
+        obsolete = nil
+    },
+}
+
+for _, option in ipairs(compat_options) do
+    local option_name = 'box_cfg_' .. option.name
+    compat.add_option({
+        name = option_name,
+        default = option.obsolete and 'new' or 'old',
```

Comment:
Those are three stages:

* old by default
* new by default
* obsolete (always new)

Reply:
Judging by `compat.lua`, one sets `option.default` to 'new' and `option.obsolete` to true at the same time, to make the option "obsolete". Am I missing something here?
https://github.com/tarantool/tarantool/blob/db6245f8b9d2720fb585cb16a6af8175d50c1bd2/src/lua/compat.lua#L158-L170

Reply:
Never mind. I see what you mean.
The option may be "new" by default, but not yet obsolete.

Reply:
Fixed:
```diff
diff --git a/src/box/lua/load_cfg.lua b/src/box/lua/load_cfg.lua
index 6bed23cad..0fca98f5b 100644
--- a/src/box/lua/load_cfg.lua
+++ b/src/box/lua/load_cfg.lua
@@ -218,7 +218,8 @@ local compat_options = {
         brief = replication_sync_timeout_brief,
         oldval = 300,
         newval = 0,
-        obsolete = nil
+        obsolete = nil,
+        default = 'old',
     },
 }
 
@@ -226,7 +227,7 @@ for _, option in ipairs(compat_options) do
     local option_name = 'box_cfg_' .. option.name
     compat.add_option({
         name = option_name,
-        default = option.obsolete and 'new' or 'old',
+        default = option.default,
         obsolete = option.obsolete,
         brief = option.brief,
         action = function(is_new)
```